pex/vendor/__main__.py

```diff
@@ -105,14 +105,33 @@ def _modify__import__calls(self, red_baron):  # noqa: We want __import__ as part
         first_argument = call_node[0]
         raw_value = self._find_literal_node(parent, first_argument)
         if raw_value:
-          value = raw_value.to_python()
+          original_value = raw_value.to_python()
+          value = original_value
           root_package = value.split('.')[0]
           if root_package in self._packages:
-            raw_value.replace('{!r}'.format(self._prefix + '.' + value))
+            value = self._prefix + '.' + value
 
+          value = self._revendor(value)
+          if value != original_value:
+            raw_value.replace('{!r}'.format(value))
             parent.replace(self._modify_import(original, parent))
             yield original, parent
 
+  def _revendor(self, value):
+    """Re-vendors transitively vendored imports to our vendored versions.
+
+    Some of our transitively vendored imports are incredibly expensive to import. Rather than
+    importing multiple versions of them, we canonicalize all of these imports to the version we
+    ourselves vendor, so that we only have to process the imports one time.
+    """
+    return (
+      value
+        .replace("pex.third_party.setuptools.extern", "pex.third_party")
```

Comment:
It would be really great if `_revendor` or some higher level that passed it in this configuration data also nuked the revendered packages. That would reduce deployed Pex PEX and Pex .whl sizes and perhaps be less confusing. Even better would be nuking and dropping in a small auto-generated README explaining the package was nuked and replaced with `VendorSpec.pinned('six', '1.14.0')` for example.